tests/types/test_schema.py

```diff
@@ -667,3 +668,102 @@ def test_enforce_tensor_spec_variable_signature():
         match=re.escape(r"Shape of input (2,) does not match expected shape (-1, 2, 3)."),
     ):
         _enforce_tensor_spec(ragged_array, standard_spec)
+
+
+def test_datatype_is_instance():
+    assert DataType.is_instance("string", DataType.string)
+
+    assert DataType.is_instance(1, DataType.integer)
```

Comment:
Can we add a method named `equal` that works like this?
 
```
DataType.integer.equal(1)  # true
```

Reply:
`equal` actually is confusing.

Reply:
Another approach is implement a method like `is_string`.


`DataType.is_string("str")` # true